runtime/container.go

```diff
@@ -37,35 +39,57 @@ type Command interface {
 	Wait() error
 }
 
-func newContainer(c libcontainer.Container, p *Process) Container {
+func newContainer(c libcontainer.Container, p *Process, imageDesc *v1.Image) Container {
 	cp := libcontainer.Process(*p)
 	return &container{
-		Container: c,
-		process:   &cp,
+		Container:       c,
+		process:         &cp,
+		imageDescriptor: imageDesc,
 	}
 }
 
 type container struct {
 	libcontainer.Container
-	process *libcontainer.Process
+	process         *libcontainer.Process
+	imageDescriptor *v1.Image
 }
 
 func (c *container) Start() error {
+	env := make([]string, len(c.imageDescriptor.Config.Env))
+	copy(env, c.imageDescriptor.Config.Env)
+	c.process.Env = append(env, c.process.Env...)
+
 	if err := c.Container.Run(c.process); err != nil {
-		c.Container.Destroy()
+		_ = c.Container.Destroy()
 		return err
 	}
 
 	return nil
 }
 
 func (c *container) Wait() error {
+	var derr error
+	defer func() {
+		derr = c.Container.Destroy()
+		streams := []interface{}{
+			c.process.Stdin,
+			c.process.Stdout,
+			c.process.Stderr,
+		}
+		for _, s := range streams {
+			if c, ok := s.(io.Closer); ok {
+				if err := c.Close(); err != nil && derr != nil {
```

Comment:
If there are multiple errors, only the first one is returned. So it does not override the error if it is already set.